### PR TITLE
feat: add halo2-wasm cli

### PR DESCRIPTION
**Pull request by rpalakkal**

### CODE REVIEW

File: cli/src/verify.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { CircuitScaffold } from "./scaffold";
+import { getFunctionFromTs, getRunCircuitFromTs, getUint8ArrayFromBuffer, readJsonFromFile } from "./utils";
+
+export const verify = async (path: string, options: { vk: string, proof: string, instances: string, circuit: string }) => {
```
Comment (jonathanpwang) [inline cli/src/verify.ts:4]: To verify shouldn't you only need: `vkey`, `instances`, `proof`? There should no longer be a need for `circuit`

**Review by jonathanpwang — APPROVED** [review]
LGTM for now, left some notes to add to issues / consider in the long-term.

For the short-term, the following should be mentioned in the readme:
* Need to use `-c src/examples/run.ts src/examples/circuit.ts` to use the default runner (consider just using default runner unless an override is provided`
* To use default runner, `halo2-lib-js` needs to be globally installed
* Global install needs to use `npm` and not `pnpm` (haven't tried `yarn`)
* `npm` global install without `sudo`: https://github.com/sindresorhus/guides/blob/main/npm-global-without-sudo.md
* Global install of `halo2-wasm-cli` to use the `halo2-wasm` command

File: cli/src/verify.ts (hunk 1 above)
Comment (rpalakkal) [reply cli/src/verify.ts:4]: reading the vkey seems to require `BaseCircuitParams`, which is instantiated in `halo2-wasm` from `CircuitConfig`. Currently the circuit file is responsible for exporting the circuit config (https://github.com/axiom-crypto/halo2-browser/blob/8737bac8ce42c2f5c9135f5f9daf482b15bbcc82/cli/src/examples/circuit.ts#L20). Will add this to the readme.

File: cli/src/prove.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { CircuitScaffold } from "./scaffold";
+import { getFunctionFromTs, getRunCircuitFromTs, getUint8ArrayFromBuffer, saveBufferToFile, saveJsonToFile } from "./utils";
+
+export const prove = async (path: string, options: { pk: string, stats: boolean, proof: string, instances:string, circuit: string }) => {
+    const circuit = await getFunctionFromTs(path);
```
[the remark this answers is not in the document]
Comment (rpalakkal) [reply cli/src/prove.ts:5]: added an `inputs` field to mock/prove commands to optionally read inputs from a json file (if it's not provided, then just defaults to the circuit file)

**Pull request merged**